src/Utilities/Compiler/Extensions/CharExtensions.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+namespace Analyzer.Utilities.Extensions
+{
+    internal static class CharExtensions
+    {
+        // From: https://github.com/dotnet/runtime/blob/d1fc57ea18ee90aee8690697caed2b9f162409eb/src/libraries/System.Private.CoreLib/src/System/Char.cs#L91
+        public static bool IsAscii(this char c) => (uint)c <= '\x007f';
```

Comment:
> public static bool IsAscii(this char c) => (uint)c <= '\x007f'; [](http://example.com/codeflow?start=7&length=64)

For the analyzer cases we are implementing here the check for ASCII should include `>= 0x20` too. The reason is, even when using Invariant culture the check like 

`"".IndexOf('\0', StringComparison.InvariantCulture)` will return 0. In other word, there are some characters in the ASCII range 0~0x1F has special treating.

Reply:
Also, would be good to change the name of this method to something like `IsNonIgnorableAsciiChar` or similar.

Reply:
Then I'll move this method to inside the analyzer and update it there, since it's not exactly ascii anymore.

Reply:
Or what you suggested 😅 A rename sounds good too.

Reply:
I am fine either way.

Reply:
I used the following range: [x0020, x007e], skipping 7f too (del). This range represents the printable ascii characters.

Reply:
That is fine. We can include `7F` too as it shouldn't affect the case as it is not ignored character. But I am fine if you want to exclude it.